practice-app/server/src/models/tweet/tweet.model.js

```diff
@@ -0,0 +1,133 @@
+                ],
+                1: [
+                    { $match: { diff: { $eq: -1 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                2: [
+                    { $match: { diff: { $eq: -2 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                3: [
+                    { $match: { diff: { $eq: -3 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                4: [
+                    { $match: { diff: { $eq: -4 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                5: [
+                    { $match: { diff: { $eq: -5 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                6: [
+                    { $match: { diff: { $eq: -6 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                7: [
+                    { $match: { diff: { $eq: -7 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                8: [
+                    { $match: { diff: { $eq: -8 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+                9: [
+                    { $match: { diff: { $eq: -9 } } },
+                    {
+                        $group: {
+                            _id: "$tag", averageLength: { $avg: "$length" }, avgNumberOfWords: { $avg: "$numberOfWords" }
+                        }
+                    },],
+            }
+        },
+    ]);
+    const res = a[0]
+    const averageLengths = []
+    const averageWordCounts = []
+    for (const minute in res) {
```

Comment:
couldnt quite understand if x is tags or minutes,

Reply:
Sorry for that terrible naming :(